kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/StatefulSessionUsageIntegrationTest.java

```diff
@@ -358,4 +363,68 @@ public void testStatefulDefault() {
         reply = ruleClient.executeCommandsWithResults(CONTAINER_ID, executionCommand);
         assertEquals(ServiceResponse.ResponseType.SUCCESS, reply.getType());
     }
+
+    @Test
+    public void testAgendaFilter() {
+        // insert and fire
+        List<Command<?>> commands = new ArrayList<Command<?>>();
+        BatchExecutionCommand executionCommand = commandsFactory.newBatchExecution(commands, KIE_SESSION);
+
+        Object person = createInstance(PERSON_CLASS_NAME, PERSON_NAME, "");
+        commands.add(commandsFactory.newInsert(person, PERSON_1_OUT_IDENTIFIER));
+        commands.add(new FireAllRulesCommand(FIRED_IDENTIFIER, -1, new RuleNameEndsWithAgendaFilter("Darth", false))); // Don't fire "Set lastname for Darth" rule
+
+        ServiceResponse<ExecutionResults> reply = ruleClient.executeCommandsWithResults(CONTAINER_ID, executionCommand);
+        assertEquals(ServiceResponse.ResponseType.SUCCESS, reply.getType());
+
+        ExecutionResults actualData = reply.getResult();
+
+        Object fired = actualData.getValue(FIRED_IDENTIFIER);
+        System.out.println("fired = " + fired + ", fired.class = " + fired.getClass());
```

Comment:
Can this be removed to avoid spamming log?

Reply:
Ouch, thank you for noticing that!